### ISSUE TITLE
Tests become broken as optional parameters are required

### ISSUE THREAD

**Issue by zaalbarxx** [php-mock/php-mock-prophecy#1]
After requiring this package in composer and running tests I noticed that many of them became broken as calls expectations changed. For example if I have a function 
`getFirstWhere($arg1, $arg2, $arg3 = [])` and I set my expectation to `$mock->getFirstWhere('arg', 'arg2')` I get the error saying that expectation was not met because there was a call to `$mock->getFirstWhere('arg', 'arg2', [])`(notice the empty array at the end). Not sure what is the reason of that. Example log:

```
  39  - it returns list of sample messages
      method call:
        - getAllWhere("client_id", 5, [])
      on Double\ClientServicesRepository\ClientServicesRepository\P82 was not expected, expected calls were:
        - getAllWhere(exact("client_id"), exact(5))

```

So basically I call in my real code `getAllWhere("client_id", 5)` but somehow it takes the optional parameters into an account.

**Comment by malkusch** [comment]
Thank you for reporting this issue. You observed a side effect of [my hack to make Prophecy support pass-by-reference](https://github.com/php-mock/php-mock-prophecy/commit/c9cd844fcf4a364d021817acaea5fc031abf7e86). It was such a hack and [upstream is reluctant to support](https://github.com/phpspec/prophecy/issues/225) that anyway. So before I dig deeper into that rabbit hole, I decided to drop pass-by-reference completely. Please update to php-mock/php-mock-prophecy-0.0.2.

If one needs mocking functions with pass-by-reference go to another framework, e.g. [php-mock-phpunit](https://github.com/php-mock/php-mock-phpunit).
